app/models/manageiq/providers/vmware/infra_manager/vm/operations.rb

```diff
@@ -3,4 +3,14 @@ module ManageIQ::Providers::Vmware::InfraManager::Vm::Operations
 
   include_concern 'Guest'
   include_concern 'Snapshot'
+
+  included do
+    supports :terminate do
+      if !supports?(:control)
+        unsupported_reason(:control)
+      elsif power_state != "off"
+        _('The VM is not powered off')
```

Comment:
NOTE I chose "not powered off" rather than "is powered on" because you also can't terminate a "suspended" vm